lib/delta-table/src/Database/Table/SQLite/Simple/Exec.hs

```diff
@@ -134,8 +134,22 @@ selectWhere expr = queryNamed . Stmt.selectWhere expr
 insertOne :: IsTableSql t => Row t -> proxy t -> SqlM ()
 insertOne row proxy = executeOne (Stmt.insertOne proxy) row
 
+-- | Insert many rows into a database table.
+--
+-- As an optimization, we use a single prepared SQL statement.
 insertMany :: IsTableSql t => [Row t] -> proxy t -> SqlM ()
-insertMany rows proxy = for_ rows (`insertOne` proxy)
+insertMany rows proxy =
+    ReaderT $ \conn -> do
+        -- The 'query' string contains '?' which will be substituted
+        -- for the values in the row.
+        let query = fst $ renderStmt $ Stmt.insertOne proxy
```

Comment:
If there shouldn't be any bindings, then perhaps
```suggestion
        let (query, _noBindings) = $ renderStmt $ Stmt.insertOne proxy
```
unless we even want it to crash if there are?
```suggestion
        let (query, []) = $ renderStmt $ Stmt.insertOne proxy
```

Reply:
`_noBindings` sounds good to me.

I don't think that `let (query, []) = …` actually crashes — `let` bindings introduce [irrefutable pattern][1].

  [1]: https://www.haskell.org/tutorial/patterns.html#tut-lazy-patterns